src/network/networkd-wwan.c

```diff
@@ -0,0 +1,666 @@
+
+        assert(link);
+        assert(IN_SET(family, AF_INET, AF_INET6));
+        assert(addr);
+
+        if (!in_addr_is_set(family, addr))
+                return 0;
+
+        /* prefixlen is not checked when parsed DBus message. Let's check it here. */
+        if (prefixlen > (family == AF_INET ? 32 : 128)) {
+                log_link_debug(link,
+                               "Bearer has invalid prefix length %u for %s address, ignoring.",
+                               prefixlen, family == AF_INET ? "IPv4" : "IPv6");
+                return 0;
+        }
+
+        r = address_new(&address);
+        if (r < 0)
+                return log_oom();
+
+        address->source = NETWORK_CONFIG_SOURCE_MODEM_MANAGER;
+        address->family = family;
+        address->in_addr = *addr;
+        address->prefixlen = prefixlen;
+
+        if (address_get(link, address, &existing) < 0) /* The address is new. */
+                link->bearer_configured = false;
+        else
+                address_unmark(existing);
+
+        r = link_request_address(link, address, &link->bearer_messages,
+                                 bearer_address_handler, NULL);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "Failed to request address provided by bearer: %m");
+
+        return 0;
+}
+
+static int bearer_route_handler(
+                sd_netlink *rtnl,
+                sd_netlink_message *m,
+                Request *req,
+                Link *link,
+                Route *route) {
+
+        int r;
+
+        assert(link);
+
+        r = route_configure_handler_internal(m, req, route);
+        if (r <= 0)
+                return r;
+
+        if (link->bearer_messages == 0) {
+                link->bearer_configured = true;
+                link_check_ready(link);
+        }
+
+        return 0;
+}
+
+static int link_request_bearer_route(
+                Link *link,
+                int family,
+                const union in_addr_union *gw,
+                const union in_addr_union *prefsrc) {
+
+        _cleanup_(route_unrefp) Route *route = NULL;
+        Route *existing;
+        int r;
+
+        assert(link);
+        assert(IN_SET(family, AF_INET, AF_INET6));
+        assert(gw);
+        assert(prefsrc);
+
+        if (!in_addr_is_set(family, gw))
+                return 0;
+
+        /*
+         * FIXME: Allow not setting the default route via
+         * DHCPv4.UseGateway option
+         */
+        if (link->network->dhcp_use_gateway == 0)
+                return 0;
+
+        r = route_new(&route);
+        if (r < 0)
+                return log_oom();
+
+        route->source = NETWORK_CONFIG_SOURCE_MODEM_MANAGER;
+        route->family = family;
+        route->nexthop.family = family;
+        route->nexthop.gw = *gw;
+        if (link->network->priority != LINK_BRIDGE_PORT_PRIORITY_INVALID) {
+                route->priority = link->network->priority;
+                route->priority_set = true;
```

Comment:
Hmm? We usually do not set route priority based on the bridge port priority.
Are there any reason to do that here?